src/komponenter/krr-melding/krr-melding.tsx

```diff
@@ -5,38 +5,28 @@ import { Normaltekst } from 'nav-frontend-typografi';
 import './krr-melding.less';
 import { difiLenke } from '../../innhold/lenker';
 import tekster from '../../tekster/tekster';
-import { klikkPaDifiLenke } from '../../metrics/metrics';
-import { useContext } from 'react';
-import { BrukerregistreringContext } from '../../ducks/brukerregistrering';
-import { BrukerInfoContext } from '../../ducks/bruker-info';
 import { OppfolgingContext } from '../../ducks/oppfolging';
+import { loggAktivitet } from '../../metrics/metrics';
+import { AmplitudeAktivitetContext } from "../../ducks/amplitude-aktivitet-context";
 
 const KrrMelding = () => {
-    const brukerregistreringData = useContext(BrukerregistreringContext).data;
-    const brukerinfoData = React.useContext(BrukerInfoContext).data;
     const oppfolgingData = React.useContext(OppfolgingContext).data;
-    const {registrering} = brukerregistreringData;
-    const {besvarelse} = registrering;
-    const {dinSituasjon} = besvarelse;
-    const dinSituasjonOrIngenVerdi = dinSituasjon ? dinSituasjon : 'INGEN_VERDI';
-    const {registreringType, rettighetsgruppe} = brukerinfoData;
-    const {formidlingsgruppe, servicegruppe, underOppfolging, reservasjonKRR} = oppfolgingData;
-    const underOppfolgingJaNei = underOppfolging ? 'ja' : 'nei';
-    const reservasjonKRRJaNei = reservasjonKRR ? 'ja' : 'nei';
-    const registreringTypeOrIngenVerdi = registreringType ? registreringType : 'INGEN_VERDI';
+    const { reservasjonKRR } = oppfolgingData;
+    const amplitudeAktivitetsData = React.useContext(AmplitudeAktivitetContext);
 
-    const metrikkData = {
-        servicegruppe,
-        formidlingsgruppe,
-        rettighetsgruppe,
-        dinSituasjon: dinSituasjonOrIngenVerdi,
-        underOppfolging: underOppfolgingJaNei,
-        registreringType: registreringTypeOrIngenVerdi,
-        reservasjonKRR: reservasjonKRRJaNei,
+    const handleLenkeKlikk = () => {
+        loggAktivitet({ aktivitet: 'Går til krr-oppsett', ...amplitudeAktivitetsData })
     };
 
-    const klikkLenkeMetrikk = () => klikkPaDifiLenke(metrikkData);
-
+    React.useEffect(() => {
+        if (reservasjonKRR) {
+            loggAktivitet({ aktivitet: 'Viser krr melding', ...amplitudeAktivitetsData });
```

Comment:
Jeg er litt usikker på hvordan livssyklusen blir her, men vil denne loggmeldingen noen sinne trigge, når default-verdien til `reservasjonKRR` er `false`? Siden vi ikke har `reservasjonKRR` i listen over avhengigheter, vil den ikke forsøke å logge på nytt når den potensielt blir satt til `true` på et senere punkt. Burde vi i så fall legge den til?

Reply:
Sånn som jeg ser kommer det an på rekkefølgen av hendelser her. Tidligere har ikke komponentene rendret før datafetchingen har fullført og da kjørte `useEffect` én gang med korrekt tilstand. Hvis det patternet ikke er endret bør dette gå greit.

Reply:
Ja, det kan man gjøre.
Det er vel som regel lite endringer i state for siden så den ville nok fungert som tenk uansett :-)

Reply:
oppdatert og tar nå med alle